metricflow/sql_clients/big_query.py

```diff
@@ -57,29 +67,39 @@ def from_connection_details(url: str, password: Optional[str]) -> BigQuerySqlCli
 
         return BigQuerySqlClient(password=password)
 
-    def __init__(self, project_id: str = "", password: Optional[str] = None, query_string: str = "") -> None:
+    def __init__(self, project_id: str = "", password: Optional[str] = None) -> None:
         """Creates a new BigQueryDBClient and tags it for tracing as big query."""
         # Without pool_pre_ping, it's possible for timed-out connections to be returned to the client and cause errors.
         # However, this can cause increase latency for slow engines.
         self._password = password
-        self._project_id = project_id
-        super().__init__(
-            engine=BigQuerySqlClient._create_bq_engine(
-                query_string=query_string, project_id=project_id, password=password
-            )
+        password_json = json.loads(password) if password else {}
+        self._project_id = project_id or password_json.get("project_id")
+        bq_engine = BigQuerySqlClient._create_bq_engine(project_id=project_id, password=password)
+        self._bq_client = bigquery.Client(
+            project=self._project_id,
+            credentials=google.oauth2.service_account.Credentials.from_service_account_info(password_json),
```

Comment:
Why are all of these changes in this commit?

Also, does this work if we aren't using service accounts?

Reply:
Yeah, some of these changes could have been a separate PR, but saw some things that could use fixing.

Not entirely sure about initialization without credentials, but this is the way that the docs create a client: https://github.com/googleapis/python-bigquery/blob/HEAD/samples/client_query.py